stdlib/native.rb

```diff
@@ -470,8 +470,16 @@ def initialize(defaults = undefined, &block)
 
           if (value && value.constructor === Object) {
             smap[key] = #{Hash.new(`value`)};
-          }
-          else {
+          } else if (value && value.constructor === Array) {
```

Comment:
can this be `(value && value.$$is_array)` instead? 

(see https://github.com/opal/opal/blob/master/opal/corelib/array.rb#L8)